include/oneapi/dpl/pstl/hetero/algorithm_ranges_impl_hetero.h

```diff
@@ -585,17 +585,33 @@ __pattern_reduce_by_segment(_ExecutionPolicy&& __exec, _Range1&& __keys, _Range2
     //          __out_keys   = { 1, 2, 3, 4, 1, 3, 1, 3, 0 }
     //          __out_values = { 1, 2, 3, 4, 2, 6, 2, 6, 0 }
 
-    if (__keys.empty())
+    const auto __n = __keys.size();
+
+    if (__n == 0)
         return 0;
 
+    if (__n == 1)
+    {
+        __brick_copy<_ExecutionPolicy> __copy_range{};
+
+        oneapi::dpl::__internal::__ranges::__pattern_walk_n<2>(::std::forward<_ExecutionPolicy>(__exec), __copy_range,
```

Comment:
Why constants are 2 and 3?

Reply:
There was linking error with values 1 and 2